sphinx_view/builders.py

```diff
@@ -0,0 +1,17 @@
+from sphinxcontrib.serializinghtml import JSONHTMLBuilder
+from sphinx.environment.adapters.toctree import TocTree
+
+
+class SphinxViewBuilder(JSONHTMLBuilder):
+
+    name = 'json'  # we're going to override the default
+
+    def get_doc_context(self, docname, body, metatags):
```

Comment:
Just an idea here (I have no idea yet... 😜)

Does `metatags` here have access to _[Field lists](https://www.sphinx-doc.org/en/master/usage/restructuredtext/field-lists.html)_?

Could we add a `:template: no-globaltoc.html` say to allow us to choose the template name later on?

Reply:
So it appears [`metatags` is for generating `<meta>`](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#html-metadata) HTML markup using the `meta` directive, which expects certain fieldnames such as "description" and "keywords".

It's already served in the `doc` block that I added that to the template context and was able to render something like `<meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />`.

So no, it appears `metatags` here does not have access to *Field lists*.

HOWEVER

The superclass's `get_doc_context` method [does indeed have access to document metadata.](https://github.com/sphinx-doc/sphinx/blob/b8789b4cb6ca034ed7209b9adc91095b0b9b763c/sphinx/builders/html/__init__.py#L595)

So we would need to:

1. Add `:template: no-globaltoc.html` to each .rst file
2. Then grab it in our `get_doc_context` with `out_dict["meta"]["template"]`

I'm not sure where to go from there, but I believe this would do what you're asking about!

Reply:
> Add :template: no-globaltoc.html to each .rst file

Only for those **not** using the default. 


SO... the idea is that we'd generate the doc JSON (see [get_context_data](https://github.com/carltongibson/django-sphinx-view/blob/53684cc6d0c0cd9f5fb727ec671420268310d8cd/sphinx_view/views.py#L33) — and either pass that into the template as part of the context (and use that in the template to decide, say, the base template) or we'd do that earlier and use it in `get_template_names()`.

 I need to have a little think about the best API there. Without really considering it I suspect the latter `get_template_names()` approach would be more natural. 

SO... We should get this in ≈as it is, and then work on that in a separate PR. Is there anything left from your POV here? (If not I'll give it the final look over this weekend.)

Thanks for the input @lancegoyke!

Reply:
I'll be curious to see how this transpires.

I do believe we have a basic version working now. Let me know if you need anything else on this!

Thanks for all that you do @carltongibson!